pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java

```diff
@@ -619,17 +629,21 @@ public void removeProducer(Producer producer) {
 
     protected void handleProducerRemoved(Producer producer) {
         // decrement usage only if this was a valid producer close
-        long newCount = USAGE_COUNT_UPDATER.decrementAndGet(this);
-        if (newCount == 0) {
+        USAGE_COUNT_UPDATER.decrementAndGet(this);
```

Comment:
Since we add an issue withe consumers here, can you add a test with consumer, to make sure we don’t break it in the future?

Reply:
sure